lxd/db/cluster/secrets.go

```diff
@@ -0,0 +1,135 @@
+
+// AuthSecretValue is used to byte slice of length authSecretLength in a text column as a base64 encoded string.
+type AuthSecretValue []byte
+
+// newAuthSecretValue returns a AuthSecretValue, read from [rand.Reader].
+func newAuthSecretValue() AuthSecretValue {
+	secretValue := make([]byte, authSecretLength)
+	_, _ = rand.Read(secretValue)
+	return secretValue
+}
+
+// Validate enforces that the AuthSecretValue has length authSecretLength.
+func (s AuthSecretValue) Validate() error {
+	if len(s) != authSecretLength {
+		return fmt.Errorf("Secret must have length %d", authSecretLength)
+	}
+
+	return nil
+}
+
+// Value implements [driver.Valuer] for AuthSecretValue.
+func (s AuthSecretValue) Value() (driver.Value, error) {
+	err := s.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	return base64.StdEncoding.EncodeToString(s), nil
+}
+
+// ScanText implements [query.TextScanner] for AuthSecretValue to simplify the [sql.Scanner] implementation.
+func (s *AuthSecretValue) ScanText(str string) error {
+	out, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		return err
+	}
+
+	*s = out
+	return s.Validate()
+}
+
+// Scan implements [sql.Scanner] for AuthSecretValue.
+func (s *AuthSecretValue) Scan(value any) error {
+	return query.ScanValue(value, s, false)
+}
+
+// SecretType represents the "type" column in the secrets table.
+type SecretType string
+
+const (
+	// SecretTypeClusterAuth is the SecretType for global auth secrets.
+	SecretTypeClusterAuth SecretType = "cluster_auth"
+)
+
+const (
+	// secretTypeCodeClusterAuth is the database code for SecretTypeClusterAuth.
+	secretTypeCodeClusterAuth int64 = 1
+)
+
+// Value implements [driver.Valuer] for SecretType.
+func (s SecretType) Value() (driver.Value, error) {
+	switch s {
+	case SecretTypeClusterAuth:
+		return secretTypeCodeClusterAuth, nil
+	}
+
+	return nil, fmt.Errorf("Invalid secret type %q", s)
+}
+
+// ScanInteger implements [query.IntegerScanner] for SecretType to simplify the [sql.Scanner] implementation.
+func (s *SecretType) ScanInteger(code int64) error {
+	switch code {
+	case secretTypeCodeClusterAuth:
+		*s = SecretTypeClusterAuth
+	}
+
+	return fmt.Errorf("Invalid secret type code %d", code)
+}
+
+// Scan implements sql.Scanner for SecretType.
+func (s *SecretType) Scan(value any) error {
+	return query.ScanValue(value, s, false)
+}
+
+// AuthSecret contains an AuthSecretValue and a creation time.
+type AuthSecret struct {
+	Value     AuthSecretValue
+	CreatedAt time.Time
+}
+
+// newAuthSecret mints a new AuthSecret.
+func newAuthSecret() AuthSecret {
+	return AuthSecret{
+		Value:     newAuthSecretValue(),
+		CreatedAt: time.Now().UTC(),
+	}
+}
+
+// IsValid returns true if the time now is less than the time that the secret was created, plus the expiry.
+func (s AuthSecret) IsValid(expirySeconds int64) bool {
+	return s.CreatedAt.Add(time.Duration(expirySeconds)*time.Second).After(time.Now().UTC()) && s.Value.Validate() == nil
```

Comment:
As we already have `Validate` for the type above, `IsValid` seems to stand out as similar in name, but quite different in purpose, how about we call this `Expired()`?

Reply:
Changed to `IsExpired` and changed all `if IsValid() { ... }` to `if !IsExpired() { ... }`.